python/pyspark/sql/session.py

```diff
@@ -276,8 +315,20 @@ def getOrCreate(self) -> "SparkSession":
                     ).applyModifiableSettings(session._jsparkSession, self._options)
                 return session
 
-    builder = Builder()
-    """A class attribute having a :class:`Builder` to construct :class:`SparkSession` instances."""
+    # TODO: This workaround can be dropped once support for Python 3.8 is dropped
+    # In Python 3.9, the @property decorator has been made compatible with the @classmethod decorator.
+    # https://docs.python.org/3.9/library/functions.html#classmethod
+    if sys.version >= "3.9":
+        @classmethod
+        @property
+        def builder(cls):
+            """A class attribute having a :class:`Builder` to construct :class:`SparkSession` instances."""
+            return cls.Builder()
+    else:
+        @classproperty
+        def builder(cls):
+            """A class attribute having a :class:`Builder` to construct :class:`SparkSession` instances."""
+            return cls.Builder()
```

Comment:
Let's don't bother w/ the old codes, and just leave it as is. We will likely drop Python 3.8 in the near future in any event.

Reply:
If I do that, the doctests won't pass in Python 3.8. 
Remember this is a bugfix, I'd rather have pySpark behave the same way with every Python version.

Reply:
Yeah, that's fine. You can just make the doctests skipped with ` # doctest: +SKIP`. I worry if we introduce another bug by mimicking this class property vs the standard implementation in Python 3.9. e.g.) we should check how it works if somebody manually sets an attribute here.

Reply:
Actually, it's a little difficult to drop Python 3.8 because the end of support is 2024/10.
- https://www.python.org/downloads

Python 3.7 also is still used frequently like in Google Colab because the end of support is 2023-06-27.
<img width="253" alt="Screen Shot 2022-04-14 at 3 35 06 PM" src="https://user-images.githubusercontent.com/9700541/163488911-313a3df5-3d85-465f-9ecb-bc28c6d671cf.png">